### PR TITLE
spg chapter10

### PR DESCRIPTION
**Pull request by SimonGross**
Chapter 10

### CODE REVIEW

File: chapter10/spg/exercise1.pl [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+print "Files in dir:", map { " $_" } @files, "\n";
+
+# Power of the Schwartz
+my @sorted =
+  map { $_->[0] }
+  sort { $a->[1] <=> $b->[1] }
+  map [ $_, get_size( $_ ) ],
+  @files;
+
+print "Files sorted by size:", map { " $_" } @sorted;
```
Comment (alexdeas) [inline chapter10/spg/exercise1.pl:21]: Perl adds spaces if you interpolate a list so you don't need to map (or join), i.e.:

``` perl
print "Files: @sorted\n";
```

File: chapter10/spg/exercise2.pl [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#! /usr/bin/perl
+
+use strict;
+use warnings;
+use Benchmark qw(:all);
+
+my @files = glob '*';
```
Comment (alexdeas) [inline chapter10/spg/exercise2.pl:7]: I suppose this evens out the sort performance comparison, although means your functions need to be supplied with these args.

File: chapter10/spg/exercise4.pl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  if ( -f $path or -l $path ) { # files or symbolic links
+    return undef;
+  }
+  if ( -d $path ) {
+    my %directory;
+    opendir PATH, $path or die "Cannot opendir $path: $!";
+    my @names = readdir PATH;
+    closedir PATH;
+    for my $name ( @names ) {
+      next if $name eq '.' or $name eq '..';
```
Comment (alexdeas) [inline chapter10/spg/exercise4.pl:38]: Could filter these out with a grep before the loop.

File: chapter10/spg/exercise2.pl (hunk 2 above)
Comment (utopia-2) [inline chapter10/spg/exercise2.pl:7]: Or let your functions be closures :smile:.

**Pull request merged**